pymatgen/core/composition.py

```diff
@@ -572,6 +575,59 @@ def to_data_dict(self):
                 "elements": self.as_dict().keys(),
                 "nelements": len(self.as_dict().keys())}
 
+    def charge_balance_combos(self, oxidation_override=None):
```

Comment:
Suggestions:
i. Rename to "get_oxi_state_guesses".
ii. kwargs to have oxi_states_override and target_charge=0.
iii. Allow to find solutions that are not charge balanced (by setting sum.
iv. Use not just common_oxidation_states but all oxidation states, but rank those with only common oxidation states higher.
v. Among those that comprise only common oxidation states, rank those that have the lowest mean_square higher. This biases solutions to find low oxidation states solutions, which I think is more physical.